aws/aws.go

```diff
@@ -215,5 +246,12 @@ func V2ConfigFromURLParams(ctx context.Context, q url.Values) (awsv2.Config, err
 			})
 		opts = append(opts, awsv2cfg.WithEndpointResolverWithOptions(customResolver))
 	}
+	if dualStack {
```

Comment:
Hmm if you're just going to add them down here, I think it was better before, doing it right after the parameter was processed. I thought maybe they needed to be set in the awsv2.Endpoint like HostnameImmutable, but if that's not the case then my apologies, please change it back.

Reply:
Yes, interestingly enough, they aren't passed in with the resolver.